### ISSUE TITLE
Problem with subaggregations

### ISSUE THREAD

**Issue by adrian83** [sksamuel/elastic4s#857]
Hi, 
I'm trying to build query with nested aggregations. It is something similar to this query:

```
 {  
  "size":0,
  "query":{...},
  "aggs":{  
    "group_by_field_1":{  
      "terms":{  
        "field":"field1"
      },
      "aggs":{  
        "group_by_field_2":{  
          "terms":{  
            "field":"field2",
            "min_doc_count":2
          },
          "aggs":{  
            "tops":{  
              "top_hits":{  
                "size":100
}}}}}}}}
```

From what i see, nesting 'top_hits' aggregation is currently unsupported.  

It can be fixed by updating com.sksamuel.elastic4s.http.search.aggs.AggregationBuilderFn to this form:
```
object AggregationBuilderFn {
  def apply(agg: AggregationDefinition): XContentBuilder = {
    val builder = agg match {
     ...
      case agg: TopHitsAggregationDefinition => TopHitsAggregationBuilder(agg)
    }
    builder
  }
}
```

Unfortunately generated query contains empty aggregation at the end. 
Problem is that aggregations like 'top_hits' cannot contain nested aggregations.

```
{...
  "aggs":{
      ...
      },
      "aggs":{
        "tops":{
          "top_hits":{
            "size":100
          },
          "aggs":{}
}}}}}
```

My scala code looks like that:

```
    ...query(...).aggregations(
          termsAggregation("group_by_field_1")
            .field("field1")
            .subAggregations(termsAggregation("group_by_field_2")
              .field("field2")
              .minDocCount(days))
              .subAggregations(topHitsAggregation("tops").size(100)))
```

**Comment by sksamuel** [comment]
I'm on holiday so PR welcome  :)

**Comment by sksamuel** [comment]
Fixed in 5.30
